From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Iv=C3=A1n=20Briano?= <ivan.briano@intel.com>
Date: Wed, 22 Oct 2025 10:23:33 -0700
Subject: [PATCH]  anv: fix FS output <-> attachment map building

The arrays is first memset to OUTPUT_DISABLED, but if we iterate over
MAX_RTS instead of the actual attachment count, we end up resetting any
values not set by the application to the, probably identity, that comes
from the state.
---
 src/intel/vulkan/anv_pipeline.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/intel/vulkan/anv_pipeline.c b/src/intel/vulkan/anv_pipeline.c
index 82e56079e40c..bc2e0d124675 100644
--- a/src/intel/vulkan/anv_pipeline.c
+++ b/src/intel/vulkan/anv_pipeline.c
@@ -2988,7 +2988,7 @@ anv_graphics_pipeline_emit(struct anv_graphics_pipeline *pipeline,
          /* Build a map of fragment color output to attachment */
          uint8_t rt_to_att[MAX_RTS];
          memset(rt_to_att, ANV_COLOR_OUTPUT_DISABLED, MAX_RTS);
-         for (uint32_t i = 0; i < MAX_RTS; i++) {
+         for (uint32_t i = 0; i < state->rp->color_attachment_count; i++) {
             if (state->cal->color_map[i] != MESA_VK_ATTACHMENT_UNUSED)
                rt_to_att[state->cal->color_map[i]] = i;
          }
